docs/container.md

```diff
@@ -0,0 +1,156 @@
+# Container
+
+Container is the abstraction responsible for holding all constructors and values.
+It’s the primary means by which an application interacts with Fx.
+You teach the container about the needs of your application,
+how to perform certain operations,
+and then you let it handle actually running your application.
+
+Fx does not provide direct access to the container.
+Instead, you specify operations to perform on the container
+by providing `fx.Option`s to the `fx.New` constructor.
+
+```go
+package fx
+
+type App
+  func New(opts ...Option) *App
+  func (app *App) Run()
+
+type Option
+  func Provide(constructors ...interface{}) Option
+  func Invoke(funcs ...interface{}) Option
+```
```

Comment:
I think your intention is to show some relevant functions for this section and their relevent types, but this throws me off a little because its formatted as a code block and yet isn't really valid Go syntax. I think it's fine, just something to be aware of. Is this a common format for trying to portray this kind of information?

Reply:
This formatting was borrowed from how godoc shows things:

<img width="358" alt="image" src="https://github.com/uber-go/fx/assets/41730/572a837c-e3b2-4220-80dc-44a1d3e8eecf">

I'm not tied to it, though. We can change it if you'd like.

Reply:
Ah okay! Makes sense. I think this is good then.